### PR TITLE
Admin clinic show appointment detail, confirm status and send email confirm to user

### PR DESCRIPTION
**Pull request by at-quantran2**
## Description
- Admin show appointment detail.
- Admin change appointment status.
- Admin send email confirm to user
## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

-

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

File: resources/views/admin_clinic/appointments/show.blade.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      <label for="dob" class="col-sm-2 col-form-label font-weight-bold">@lang('admin_clinic/appointment.fields.status')</label>
+      <input id="slug" type="hidden" name="slug" value="{{ $slug }}">
+      <select id="appointment-{{ $appointment->id }}" class="col-md-4 d-inline custom-select text-body font-weight-bold status-select"
+          required name="status">
+      @foreach (App\Appointment::STATUS as $key => $status)
+          @if ($status == $appointment->status )
+          <option value="{{ $key }}" selected>{{ $status }}</option>
+          @else
+          <option value="{{ $key }}">{{ $status }}</option>
+          @endif
```
Comment (at-hainguyen5) [inline resources/views/admin_clinic/appointments/show.blade.php]: view 2 space cho dễ đọc hì
code trong `@if`, `@foreach` thì cứ indent vào giống như khi mình viết `if, foreach` bình thường

**Comment by at-hainguyen5** [comment]
LGTM

File: app/Jobs/SendAppointmentConfirmationEmail.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Support\Facades\Mail;
+use App\Mail\AppointmentConfirmationEmail;
+
+class SendAppointmentConfirmationEmail implements ShouldQueue
```
Comment (at-vienle) [inline app/Jobs/SendAppointmentConfirmationEmail.php]: laravel 5.6 hỗ trợ queue mail sẵn rồi. Không cần define ra class Job, chỉ cần 1 class Mail thôi là đủ. 
https://laravel.com/docs/5.6/mail#queueing-mail

File: app/Http/Controllers/AdminClinic/AppointmentController.php [hunk 3]
```diff
@@ -34,6 +52,7 @@ public function update(Request $request, $slug, Appointment $appointment)
     {
         if ($this->clinic->id == $appointment->clinic->id) {
             $appointment->update(['status' => $request->status]);
+            SendAppointmentConfirmationEmail::dispatch($appointment)->delay(now()->addMinutes(5))->onQueue('emails');
```
Comment (at-vienle) [inline app/Http/Controllers/AdminClinic/AppointmentController.php]: delay chi 5p dữ vậy, với những task send mail này thì ko cần delay đâu.

**Pull request merged**